src/generate-sdk/sdkGenerator/TsSdkGenerator.ts

```diff
@@ -0,0 +1,251 @@
+      const localType: TypeAlias = type as TypeAlias;
+      externalType = `${localType.definition};
+
+%%%type%%%`;
+    }
+
+    this.classImplementation = this.classImplementation.replace(
+      "%%%type%%%",
+      externalType
+    );
+  }
+
+  getMethodImplementation(
+    classDefinition: ClassDefinition,
+    methodDefinition: MethodDefinition
+  ): string {
+    let func;
+
+    if (methodDefinition.params.length === 0) {
+      const parameters = methodDefinition.params
+        .map(
+          (e: ParameterDefinition) =>
+            e.name + ": " + this.getParamType(e.paramType)
+        )
+        .join(", ");
+      const returnType = this.getReturnType(methodDefinition.returnType);
+      func = `static async ${methodDefinition.name}(${parameters})${returnType} {
+        return await ${classDefinition.name}.remote.call("${classDefinition.name}.${methodDefinition.name}")  
+  }
+
+  %%%func%%%`;
+    } else {
+      const parameters = methodDefinition.params
+        .map(
+          (e: ParameterDefinition) =>
+            e.name +
+            ": " +
+            this.getParamType(e.paramType) +
+            (e.defaultValue
+              ? " = " +
+                ((typeof e.defaultValue === "string" ? '"' : "") +
+                  e.defaultValue.toString()) +
+                (typeof e.defaultValue === "string" ? '"' : "")
+              : "")
+        )
+        .join(", ");
+      const returnType = this.getReturnType(methodDefinition.returnType);
+
+      func = `static async ${
+        methodDefinition.name
+      }(${parameters})${returnType} {
+        return await ${classDefinition.name}.remote.call("${
+        classDefinition.name
+      }.${methodDefinition.name}", ${methodDefinition.params
+        .map((e) => e.name)
+        .join(", ")})  
+  }
+
+  %%%func%%%`;
+    }
+
+    return func;
+  }
+
+  getReturnType(returnType: TypeDefinition | undefined): string {
+    if (!returnType) {
+      return "";
+    }
+
+    let value = this.getParamType(returnType);
+    if (!value.includes("Promise")) {
+      value = `Promise<${value}>`;
+    }
+
+    return `: ${value}`;
+  }
+
+  getParamType(elem: TypeDefinition): string {
+    if (elem.type === AstNodeType.NativeType) {
+      const localElem: NativeType = elem as NativeType;
+      return localElem.paramType;
+      // switch (localElem.paramType) {
+      //   case NativeTypeEnum.string:
+      //     return "String";
+      //   case NativeTypeEnum.number:
+      //     return "Double";
+      //   case NativeTypeEnum.boolean:
+      //     return "Bool";
+      //   case NativeTypeEnum.any:
+      //     return "Any";
+      //   default:
+      //     return "Any";
+      // }
+    } else if (elem.type === AstNodeType.TypeAlias) {
+      const localElem: TypeAlias = elem as TypeAlias;
+      // this.generateExternalTypes(localElem);
+
+      return localElem.name;
+    }
+
+    return "any";
+  }
+}
+
+
+const supportedLanguages = ["ts", "typescript"];
```

Comment:
`ts` and `typescript`?

Reply:
so that the user can put in the config whatever he wants